### PR TITLE
Add tokeniser to step

### PR DESCRIPTION
**Pull request by gabechu**
### Description
Create a pipeline step to get tokeniser.

#### Checklist
- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

### CODE REVIEW

**Comment by gabechu** [comment]
[Add step for tokeniser](https://trello.com/c/qYqeITER/21-add-step-for-tokeniser)

File: pii_recognition/evaluation/pipeline_test.py [hunk 1]
```diff
@@ -26,9 +27,23 @@ def mock_registry():
 
 @patch("pii_recognition.evaluation.pipeline.recogniser_registry", new=mock_registry())
 def test_get_recogniser():
-    actual = get_recogniser("RegistryNoConfig")["recogniser"]
+    tokeniser_setup = {"name": "RegistryNoConfig"}
+    actual = get_recogniser(tokeniser_setup)["recogniser"]  # it's in meta
+    assert isinstance(actual, RegistryNoConfig)
+
+    tokeniser_setup = {"name": "RegistryWithConfig", "config": {"param_a": "value_a"}}
+    actual = get_recogniser(tokeniser_setup)["recogniser"]  # it's in meta
```
Comment (gabechu) [inline pii_recognition/evaluation/pipeline_test.py:35]: `mypy` would raise error on this
`Dict entry 1 has incompatible type "str": "Dict[str, str]"; expected "str": "str"`

File: pii_recognition/evaluation/pipeline_test.py [hunk 2]
```diff
@@ -26,9 +27,23 @@ def mock_registry():
...
+
+@patch("pii_recognition.evaluation.pipeline.tokeniser_registry", new=mock_registry())
+def test_get_tokeniser():
+    tokeniser_setup = {"name": "RegistryNoConfig"}
+    actual = get_tokeniser(tokeniser_setup)
     assert isinstance(actual, RegistryNoConfig)
 
-    actual = get_recogniser("RegistryWithConfig", {"config_a": "attr_a"})["recogniser"]
+    tokeniser_setup = {"name": "RegistryWithConfig", "config": {"param_a": "value_a"}}
+    actual = get_tokeniser(tokeniser_setup)
```
Comment (gabechu) [inline pii_recognition/evaluation/pipeline_test.py:47]: `mypy` would raise error on this
`Dict entry 1 has incompatible type "str": "Dict[str, str]"; expected "str": "str"`

File: pii_recognition/evaluation/pipeline.py [hunk 3]
```diff
@@ -1,23 +1,28 @@
...
-) -> Dict[str, EntityRecogniser]:
+def get_recogniser(recogniser_setup: Dict) -> Dict[str, EntityRecogniser]:
     recogniser_instance = recogniser_registry.create_instance(
-        recogniser_name, recogniser_config
+        recogniser_setup["name"], recogniser_setup.get("config")
     )
     return {"recogniser": recogniser_instance}
 
 
+def get_tokeniser(tokeniser_setup: Dict):
```
Comment (ePak) [inline pii_recognition/evaluation/pipeline.py:20]: Curious why `get_recogniser()` puts the recogniser into meta but this one doesn't

Comment (gabechu) [reply pii_recognition/evaluation/pipeline.py:20]: It will consumed by the next step when building the evaluator and it won't be used anywhere else.

Comment (gabechu) [reply pii_recognition/evaluation/pipeline.py:20]: Evaluator used to take `tokeniser_setup` dict when initialising and then build tokeniser in-place. In one of the past PRs, we've changed to have `Evaluator` takes `tokeniser` instance for initialisation.

Comment (ePak) [reply pii_recognition/evaluation/pipeline.py:20]: For consistency, you could put it in meta as well and let pakkr injects it.

File: pii_recognition/evaluation/pipeline_test.py (hunk 1 above)
Comment (gabechu) [reply pii_recognition/evaluation/pipeline_test.py]: Reassignment causes the issue.

File: pii_recognition/evaluation/pipeline_test.py (hunk 2 above)
Comment (gabechu) [reply pii_recognition/evaluation/pipeline_test.py]: Reassignment causes the issue.

**Pull request merged**